client/state/data-layer/wpcom/sites/blog-stickers/remove/index.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal Dependencies
+ */
+import { SITES_BLOG_STICKER_REMOVE } from 'state/action-types';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { addBlogSticker } from 'state/sites/blog-stickers/actions';
+import { errorNotice, successNotice } from 'state/notices/actions';
+
+export function requestBlogStickerRemove( { dispatch }, action ) {
+	dispatch(
+		http( {
+			method: 'POST',
+			path: `/sites/${ action.payload.blogId }/blog-stickers/remove/${ action.payload.stickerName }`,
+			body: {}, // have to have an empty body to make wpcom-http happy
+			apiVersion: '1.1',
+			onSuccess: action,
+			onFailure: action,
+		} )
+	);
+}
+
+export function receiveBlogStickerRemove( store, action, next, response ) {
+	// validate that it worked
+	const isRemoved = !! ( response && response.success );
+	if ( ! isRemoved ) {
+		receiveBlogStickerRemoveError( store, action, next );
+		return;
+	}
+
+	store.dispatch(
+		successNotice(
+			translate( 'The sticker {{i}}%s{{/i}} has been successfully removed.', {
+				args: action.payload.stickerName,
+				components: {
+					i: <i />,
+				},
+			} )
+		)
+	);
+}
+
+export function receiveBlogStickerRemoveError( { dispatch }, action, next ) {
+	dispatch(
+		errorNotice( translate( 'Sorry, we had a problem removing that sticker. Please try again.' ) )
+	);
+	// Revert the removal
+	next( addBlogSticker( action.payload.blogId, action.payload.stickerName ) );
```

Comment:
We are moving away from ever using `next`: https://github.com/Automattic/wp-calypso/pull/15981

Heres an [example](https://github.com/Automattic/wp-calypso/pull/15981/files#diff-38126ed4b9b0881bde0feb04d3f0e3efR18) of how to write a dispatch that skips the data-layer.

Essentially instead of `next`, you do this at the top of the file:
```
import { local } from 'state/data-layer/utils';
```

and then this where you called next:
```
dispatch( local( action ) );
```